### ISSUE TITLE
[Feature request(?)] Volumetric flask setter

### ISSUE THREAD

**Issue by 0lafe** [GTNewHorizons/GT-New-Horizons-Modpack#6021]
#### Which modpack version are you using?
2.0.8.9

#### What do you suggest instead/what changes do you propose?

If it's easy to add, it would be nice to have a block that would set flask values without player input. You can possibly need lots and lots of flasks, so setting values 16 at a time can take ages. If we could set a value on a machine and have it set them for you on the flasks.

If possible, an easier method could be to have recipes in some other machine and use circuit integrations to set useful values. Multiplies of 144 and 250 for example. 

Also, if their stack size could be increased to 64, that could be helpful for manually setting values. Possibly using many thousands of these at varying values, is painful by hand. 4xing the amount you can do would be incredibly helpful alone.

**Comment by Retster** [comment]
That feature would be wery-wery useful. And stack size 64 is also simple to implement and would help alot. QOL change.

**Comment by leagris** [comment]
Instead of adding feature, re-enable OpenBlock tanks and uses these in crafting pattern with the fluid export bus.

**Comment by Glease** [comment]
How good is OpenBlocks tanks though? Never used them before.

**Comment by Dream-Master** [comment]
they lag as hell that's why we disabled it.

**Comment by draknyte1** [comment]
Fps wise they can cause lag in large quantities, but that’s it.

**Comment by leagris** [comment]
Are the flasks able to program patterns with fluid like OpenBlock tanks does?

**Comment by Prometheus0000** [comment]
Doesn't he want them for on demand recipe amounts? Like, send it to the super tank to get a filled one, and then to the input hatch of a PA for the recipe, then back to storage. I'm not sure what you mean about the openblocks tanks and crafting patterns though. Do you have a link for how 'program patterns with fluid like OpenBlock tanks' works?

**Comment by leagris** [comment]
- Fill OpenBlock tank with the exact amount of fluid needed for recipe. Example 144l of molten rubber.
- Place filled OpenBlock tank with other items (copper wire) for recipe in the pattern grid in machine recipe mode and select a copper cable as output.
- Encode pattern.
- Place pattern in Fluid Interface/Export Bus

→ Get on-demand recipe for copper cable crafting with exact amount of 144l molten rubber and copper wire...

EC2 bus will provide the fluid amount and the items from the recipe.

**Comment by Prometheus0000** [comment]
Wouldn't the fluid export bus just export all molten rubber? And don't patterns not go into fluid export busses to begin with? And even if you put it into a fluid interface, wouldn't it just export the 144l molten rubber and not the items? Or do you need a regular interface as well, and a coprocessor on the crafting storage? Or can the fluid interfaces export items as well? And can you not use the flasks for this? Maybe adding compatibility with the flasks would be better if it isn't too hard. Can you use the universal cells? Do you need fluid in fluid cells, or can you grab it with a fluid storage bus?

**Comment by leagris** [comment]
No, the fluid interface will export items as well.

**Comment by leagris** [comment]
Here is how it works. Can be done with a volumetric flask:

### In Pattern Terminal:

Switch to Processing Patern:
Place Volumetric flask with 144L Molten Rubber and Wire
Encode Pattern

![image](https://user-images.githubusercontent.com/1111474/80890757-a8642900-8cbf-11ea-8e27-cc414e3474f4.png)

### In Fluid Interface

Place encoded pattern into the fluid interface:
![image](https://user-images.githubusercontent.com/1111474/80890802-fda03a80-8cbf-11ea-847f-d6f26fdfa23d.png)

### In assembler

Has Fluid interface will insert desired amount of molten rubber and cables into the assembler. Place a standard ME interface to recover the assembler output into the ME again so the crafting CPU detects crafting task completion.

### Conclusion

There is no need to re-enable OpenBlock tanks. The Volumetric Flask perfectly works in crafting patterns for use in the fluid interfaces.

**Comment by Prometheus0000** [comment]
Though this is only for if you can directly interface with the machine, it's still cool. But if it's in a PA, can you split the items and fluids using conduits or pipes/item pipes? I heard that import busses aren't fast enough at high levels, do you just use a conveyor to export into the interface?

I especially love how most of this discussion is about something other than what the issue is actually about. Maybe the auto-chisel could be reused? Even if it's just for specific ones, I believe 9, 18, 36, 48, 72, 144, 250, 288, 532 would cover almost all volumetric flask uses. But can the auto-chisel work that way? I'm not sure if the size is set with nbt or metadata.

**Comment by 0lafe** [comment]
Nbt iirc. @leagris the issue is I need like thousands and thousands of the 144L cell you start with. You normally have to set them by hand to 144L, which can take hours for whole setups.

Autochisel would probably work, but that seems harder than a recipe I think. I'd also recommend values such as 576, 720 and 864, as well as 500/750. When doing large PA batches, it's a nice cell reduction to use higher quantities.

**Comment by Houstonruss** [comment]
I'd be happy so see the default value at 144mb. After setting 3000 flasks i want to die.

**Comment by Prometheus0000** [comment]
So after investigating, I may have found a solution, but it's probably sorta janky-ass.
![2020-05-02_22 20 31](https://user-images.githubusercontent.com/8517039/80897074-3392ee00-8cc3-11ea-839d-a3d8feba2dd4.png)

Basically, you use the me fluid interface on a machine hull, which CAN handle both items and liquids, so they both get sent (unlike conduits/item pipes, which won't even connect). Then use conduits (didn't test gt pipes+conveyor+pump, but they'd probably work) to send to the input bus+hatch. I'm not sure what the throughput on a machine hull is though, or if there's another good block that accepts items and fluids.

**Comment by leagris** [comment]
Machine hull has max speed throughput for fluid (maybe restricted by tier dependent internal fluid storage capacity), has default slow auto-output at the designated side if powered. Sometimes it is a better choice than pipes like for large steam turbines. And yes, conveyor and pump would work.

**Comment by Prometheus0000** [comment]
The machine hull tier doesn't seem to matter (LV vs UV, no change in speed). I checked, and there seems to be no difference in speed with PA (with one machine) vs single block using this method (both UV assemblers, with UV conveyor+pump+appropriate pipe+item pipe), so long as you do a large enough batch that the PA doesn't stall. The auto-output might be slow, but using conveyors+pumps makes it not it seems.

![2020-05-02_23 41 48](https://user-images.githubusercontent.com/8517039/80898176-82925080-8cce-11ea-97bb-d00faf6d8177.png)

There might theoretically be a problem with very small pipes if it waits to export items until it finishes fluids first, but it worked fine with tiny void metal pipes and a 192 cable recipe done several times in a row (requested 1000).

Also, you can use a stack of molten rubber cells to do 64 cables at once instead of using flasks to do it one at a time, which does slow it down since the PA turns on and off. Or even 3 stacks for 192. Or 4 if you use the new terminal.

**Comment by Glease** [comment]
Just use the block variation instead of the panel variation. It can push to all 6 sides, so you can have one side with ender fluid conduit, the other side with chest/input bus (item conduits don't connect, but chests do work).

If the recipe needs more than one fluid, it will push it one after another. It will block until all fluids in current item's recipe is outputted and continue to push fluid/item for next item's recipe.

As a side note, the infamous EC2 fluid dupe bug kicks in again. If there are more than 1 tank/conduit/pipe attached, under certain circumstance it will output more fluid than consumed.

**Comment by Prometheus0000** [comment]
Glease, could you make a ticket for that? DeForce said they'd look into the EC2 stuff at some point, so I'd like to have an actual issue so they know what to look for from someone who knows exactly how to reproduce it.

**Comment by Glease** [comment]
I actually know what goes wrong and is now fiddling with the code. If I give up I'll write an issue, otherwise I'll send the PR later when I'm done.

**Comment by Glease** [comment]
@Prometheus0000 the aforementioned dupe bug is fixed by GTNewHorizons/ExtraCells2@3a7bc94

**Comment by botn365** [comment]
my problem with the fluid interface is that you can craft fluids on demand you always need it in stock
that means for evry metal fluid you need a fluid extactor and fluid bus while with flask you make a pattern for it

**Comment by draknyte1** [comment]
Decided to quickly implement this.
Available in my next build.

**Comment by GeicowithaBango** [comment]
How about a crafting recipe that stores common values. the 3x3 crafting table could hold up to 9 pre defined values for flasks. For example: 144, 250,500,750,(1000)  etc. 
It seems like an easy solution that can be nicely integrated @Dream-Master

**Comment by leagris** [comment]
How about a workbench recipe that copy flask volume in first slot to flasks stack in the second slot?

**Comment by GeicowithaBango** [comment]
even better

**Comment by draknyte1** [comment]
![image](https://user-images.githubusercontent.com/3060479/80930694-0f134080-8dad-11ea-9025-35a2f30ab941.png)
![image](https://user-images.githubusercontent.com/3060479/80930687-07539c00-8dad-11ea-994f-41a60643f9ec.png)

GUI is still WIP cause it's post-midnight. (Will finish it up tomorrow)
But it's got some preset values and a box for entering your own. Dunno what else it needs?

**Comment by Dream-Master** [comment]
If we can't update gt++ to  a newer version than 53 how we can add it ?

**Comment by GeicowithaBango** [comment]
i like @leagris approach, gameplay-wise.

**Comment by draknyte1** [comment]
There’s no reason you can’t update @Dream-Master.. if there is, it would help if you replying on discord to my queries. 🤦🏻‍♀️

Shit won’t fix itself and I won’t fix it if I don’t know what to fix. Ffs.

**Comment by GeicowithaBango** [comment]
I guess we can just add crafting recipes then......

**Comment by draknyte1** [comment]
You can’t really add a generic crafting recipe like that though. 

You’d be limited to presets only, couldn’t just copy from A->B.

**Comment by GeicowithaBango** [comment]
having a few parameters is enough, isnt it? You still can give specific values by hand, I dont see how this is argument.

**Comment by draknyte1** [comment]
The idea is to avoid setting the values for a stack+ by hand, crafting won’t resolve that fully. So no, it’s not an argument. 💁🏻‍♀️

**Comment by GeicowithaBango** [comment]
my idea works with ae2, yours doesnt.

**Comment by draknyte1** [comment]
Don't see why not?
You can set default slot with screwdriver, then just pipe them in/out.
Custom value is retained, for slot 8.
 it's just simple   L O G I C~

**Comment by GeicowithaBango** [comment]
Just makes it more difficult for the sake of being difficult. Frankly speaking, thats dispensable.

**Comment by draknyte1** [comment]
It's uhh, exactly the same as circuit programmers. 🤔 
People hate doing that via crafting.

Community already likes the idea and even the person to suggest crafting recipes approves. No point adding NEI crafting recipe bloat when it's not required.

**Comment by GeicowithaBango** [comment]
Didnt dream just pointed out that your mod wont get updated anymore? 
I dont see then why your content should be used to fix problems or follow suggestions, specifically on this gtnh github. 
Shocking, I know.

**Comment by draknyte1** [comment]
> Didnt dream just pointed out that your mod wont get updated anymore?

No, lol. He doesn't have a working build beyond that version, as I haven't given him one. :')
I just find it odd that the mod exists and people try pretend it doesn't. Why not use the content that is there and isn't going away? 🤔

**Comment by GeicowithaBango** [comment]
> I just find it odd that the mod exists and people try pretend it doesn't. Why not use the content that is there and isn't going away? 🤔

do you wanna hear the short or long story?

**Comment by draknyte1** [comment]
Long story, please. 👏

**Comment by GeicowithaBango** [comment]
hmm bigbass is watching, and i really wanna freakin curse.

**Comment by Prometheus0000** [comment]
https://www.curseforge.com/

**Comment by Dream-Master** [comment]
I am watching too ...

**Comment by GeicowithaBango** [comment]
I will not speak in bad terms. However, there a various thing I want to point out about alks behavior for the time Im playing the pack/ being on the discord, which makes it hard for me to use any of his/hers/its stuff or even writting with him/her/it on the same channel.
1.) He/she/it is an attention whore that would deliberately destroy the game balance just for making his mod more attractable. 
2.) He/she/it doesnt cooperate with gtnh-team until Bart pressures him/her/it to the limit. 
And even then he/she/it doesnt wants to accept it, and tries to surpass it by devising lies.
3.)Anti-team-player. 
4.) Thinks highly of himself while being a cunt to everyone else, I know, probably mental disorder but still....
5.)Doesnt react to feed-back.
6.)Manages to break everything he/she/it touches and doesnt accept clearly own faults.

So in conclusion, i would rather fucking not use gt++. For those who didnt realised that until now.

**Comment by GeicowithaBango** [comment]
Cuz I have to put on my fucking double seeing glasses to begin to see the amount of Bullshit he/she/it writes.

**Comment by draknyte1** [comment]
In simple response:
1) I listen to community on balance input & Dream if asked.
2) Bart doesn't do shit, nor work with me. I always cooperate with Dream. (Bart just breaks stuff)
3) I'm not part of any team, so.. irrelevant point.
4) It's the internet, might as well have a superiority complex.
5) Reacts to all constructive feedback and criticism, which people fail to provide more often than not.
6) I will if proven at fault, but generally I can prove otherwise.

Nobody really cares what you do, but speaking for yourself on a community topic isn't really the goal here now is it?

🍿

**Comment by ScriptedPiky** [comment]
First off, while Alk being an asshole is true for the most part, Alk has added three very important items that **will** disable three separate things on the multiblocks: Parallel, Speed, and Energy Reduction. Alk has even told me that you can apply this by simply making the base machine class require the 'no bonus chip' in the special-slot (hey Alk, can you confirm this?). Just wanted to point this out on the latest version of GT++ 53a and GTNH 9.0

**Comment by Dream-Master** [comment]
So i close the conversation now. This is to heated and github issues are not for this kind of discussion.

**Comment by Dream-Master** [comment]
@GTNewHorizons/core @GTNewHorizons/developers feel free to discusse about the original issue. Thanks

**Comment by Prometheus0000** [comment]
See #6733 and https://github.com/GTNewHorizons/GT5-Unofficial/pull/321 until/unless alk makes the machine fully usable (automatable)

**Comment by botn365** [comment]
with my changes this can be closed now imo

**Comment by botn365** [comment]
my bad sinse this fix is reacent should get closed after a new version is released and just mark it as fixed in dev
